test/unittests/stack_test.cpp

```diff
@@ -326,3 +327,17 @@ TEST(operand_stack, to_vector)
     EXPECT_EQ(result[1].i64, 2);
     EXPECT_EQ(result[2].i64, 3);
 }
+
+TEST(operand_stack, hidden_stack_item)
+{
+    constexpr auto max_height = 33;
+    OperandStack stack(nullptr, 0, 0, max_height);
+    ASSERT_GT(address_diff(&stack, stack.rbegin()), 100) << "not allocated on the heap";
+    EXPECT_EQ(stack.size(), 0);
+    EXPECT_EQ(stack.rbegin(), stack.rend());
+
+    // Even when stack is empty, there exists a single hidden item slot.
+    const_cast<fizzy::Value*>(stack.rbegin())->i64 = 1;
+    EXPECT_EQ(stack.rbegin()->i64, 1);
```

Comment:
Hm, I think we should document this behaviour in the header.

Reply:
No. This is a logical error to do this.

Reply:
The documentation (https://en.cppreference.com/w/cpp/iterator/begin) doesn't state anywhere that `begin()` would be UB in case of `size() == 0`.

Reply:
@chfast any feedback?

Reply:
Taking the iterator is fine, but you cannot dereference it when it is the end iterator.

Reply:
But here you are dereferencing it?

To be honest at this stage I lost what the conversation is about.

Reply:
Ah okay, we do ensure that `begin() == end()`, just that we know `begin()` is always valid.